### PR TITLE
Add Unit Tests for OrganizationAgendaCategory Component and Search Functionality for Agenda Item Categories

### LINKED ISSUE #2372: Write Unittests for `src/components/OrgSettings/AgendaItemCategories/OrganizationAgendaCategory.tsx`
We will need unittests done for all methods, classes and functions found in this file.

### PR Acceptance Criteria

1. When complete this file must show **100%** coverage when merged into the code base.
1. [The current code coverage for the file can be found here](https://app.codecov.io/gh/PalisadoesFoundation/talawa-admin/tree/develop/?displayType=list)
1. CodeRabbit must approve all your changes to the test file(s) and the file(s) under test
1. The PR will show a report for the code coverage for the file you have added. You can use that as a guide
1. No functionality must be removed to achieve this goal
1. All tests must be valid and truly test the code base

### Other Criteria

1. If this file isn't referenced in the code base then remove it. We don't want tests run on unused code

### Potential internship candidates

Please read this if you are planning to apply for a Palisadoes Foundation internship 
- https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by im-vedant**
**What kind of change does this PR introduce?**

Refactoring and feature

**Issue Number:**

Fixes #2372 

**Did you add tests for your changes?**

Yes

**Snapshots/Videos:**

https://www.loom.com/share/fb28b6228fc649158f8db6eba1ab1fc6

**Summary**
1. Unit Tests: Added unit tests for the OrganizationAgendaCategory component located in src/components/OrgSettings/AgendaItemCategories/OrganizationAgendaCategory.tsx.

2. Search Functionality: Implemented a search feature to filter agenda item categories based on user input.

## Summary by CodeRabbit

- **New Features**
	- Added search functionality for agenda item categories.
	- Enhanced filtering capabilities for agenda categories.

- **Bug Fixes**
	- Improved error handling for category-related operations.

- **Style**
	- Updated layout and styling for agenda category components.
	- Changed CSS imports and class names for better consistency.

- **Tests**
	- Added comprehensive test coverage for search functionality.
	- Implemented new test scenarios for loading and error states.

### CODE REVIEW

**Comment by palisadoes** [comment]
1. Please get the code coverage for the patch up to 100% 
2. Make CR approve your PR

File: src/components/OrgSettings/AgendaItemCategories/OrganizationAgendaCategory.module.css [hunk 1]
```diff
@@ -13,16 +13,32 @@
...
+  min-width: 18rem;
+  position: relative;
 }
 
-.btnsContainer button {
-  display: flex;
-  align-items: center;
+.btnsContainer input {
+  /* Styling for the input border */
+  outline: 1px solid var(--bs-gray-400);
```
Comment (palisadoes) [inline src/components/OrgSettings/AgendaItemCategories/OrganizationAgendaCategory.module.css:25]: Please migrate this stylesheet to:
- https://github.com/PalisadoesFoundation/talawa-admin/blob/develop-postgres/src/style/app.module.css

Use only pre-existing variables as much as possible

File: src/components/OrgSettings/AgendaItemCategories/OrganizationAgendaCategory.module.css [hunk 2]
```diff
@@ -13,16 +13,32 @@
...
 }
 
+.btnsContainer .input button {
+  /* Styling for the button in the input field */
+  width: 52px;
+}
+
+.inputField {
+  background-color: white;
+  box-shadow: 0 1px 1px #31bb6b;
```
Comment (palisadoes) [inline src/components/OrgSettings/AgendaItemCategories/OrganizationAgendaCategory.module.css:35]: 1. We are migrating away from the green theme as part of our support for color blind users.
2. Take cues from other similar elements in this file and the PR mentioned in the issue.
    1. https://github.com/PalisadoesFoundation/talawa-admin/pull/2466

**Review by palisadoes — CHANGES_REQUESTED** [review]
See comments

**Comment by palisadoes** [comment]
Comment here when this is ready for review

**Comment by im-vedant** [comment]
@palisadoes can you review this?

**Comment by palisadoes** [comment]
@coderabbitai full review

Please make coderabbitai approve your PR

**Review by palisadoes — APPROVED** [review]

**Comment by palisadoes** [comment]
@coderabbitai full review

**Pull request merged**